components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/IdentityOAuth2ScopeConsentClientException.java

```diff
@@ -0,0 +1,23 @@
+package org.wso2.carbon.identity.oauth2;
+
+/**
+ * Identity OAuth scope consent client exception.
+ */
+public class IdentityOAuth2ScopeConsentClientException extends IdentityOAuth2ScopeConsentException {
```

Comment:
Since we are having the new methods inside the oauth2scope service do we need a new exception classes? we can use IdentityOAuth2ScopeException related classes

Reply:
Since these exceptions are related OAuth scope consent management and it is handled by a different DAO layer, introduced a new exception class for the exceptions.